src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationIntTest.kt

```diff
@@ -103,10 +118,524 @@ class AdjudicationsMigrationIntTest : SqsIntegrationTestBase() {
+            status = MigrationStatus.COMPLETED,
+            estimatedRecordCount = 123_567,
+            filter = "",
+            recordsMigrated = 123_560,
+            recordsFailed = 7,
+            migrationType = MigrationType.ADJUDICATIONS,
+          ),
+        )
+      }
+    }
+
+    @AfterEach
+    fun deleteHistoryRecords() {
+      runBlocking {
+        migrationHistoryRepository.deleteAll()
+      }
+    }
+
+    @Test
+    fun `must have valid token to get history`() {
+      webTestClient.get().uri("/migrate/adjudications/history")
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `must have correct role to get history`() {
+      webTestClient.get().uri("/migrate/adjudications/history")
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_BANANAS")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `can read all records with no filter`() {
+      webTestClient.get().uri("/migrate/adjudications/history")
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ADJUDICATIONS")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.size()").isEqualTo(4)
+        .jsonPath("$[0].migrationId").isEqualTo("2020-01-03T02:00:00")
+        .jsonPath("$[1].migrationId").isEqualTo("2020-01-02T02:00:00")
+        .jsonPath("$[2].migrationId").isEqualTo("2020-01-02T00:00:00")
+        .jsonPath("$[3].migrationId").isEqualTo("2020-01-01T00:00:00")
+    }
+
+    @Test
+    fun `can filter so only records after a date are returned`() {
+      webTestClient.get().uri {
+        it.path("/migrate/adjudications/history")
+          .queryParam("fromDateTime", "2020-01-02T02:00:00")
+          .build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ADJUDICATIONS")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.size()").isEqualTo(2)
+        .jsonPath("$[0].migrationId").isEqualTo("2020-01-03T02:00:00")
+        .jsonPath("$[1].migrationId").isEqualTo("2020-01-02T02:00:00")
+    }
+
+    @Test
+    fun `can filter so only records before a date are returned`() {
+      webTestClient.get().uri {
+        it.path("/migrate/adjudications/history")
+          .queryParam("toDateTime", "2020-01-02T00:00:00")
+          .build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ADJUDICATIONS")))
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.size()").isEqualTo(2)
+        .jsonPath("$[0].migrationId").isEqualTo("2020-01-02T00:00:00")
+        .jsonPath("$[1].migrationId").isEqualTo("2020-01-01T00:00:00")
```

Comment:
Just noticed the different order here - is that on purpose?

Reply:
this looks fine - migration time descending